File: PHP/calculator.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Simple PHP Calculator</title>
</head>
<body>
<h2>Simple Calculator</h2>
<form method="post">
    <label for="num1">Number 1:</label>
    <input type="number" name="num1" id="num1" step="any" required>
    <br><br>
    <label for="num2">Number 2:</label>
    <input type="number" name="num2" id="num2" step="any" required>
    <br><br>
    <label for="operation">Operation:</label>
    <select name="operation" id="operation" required>
        <option value="add">Addition</option>
        <option value="subtract">Subtraction</option>
        <option value="multiply">Multiplication</option>
        <option value="divide">Division</option>
    </select>
    <br><br>
    <button type="submit">Calculate</button>
</form>

<?php
if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $num1 = $_POST['num1'];
    $num2 = $_POST['num2'];
    $operation = $_POST['operation'];

    $result = null;

    switch ($operation) {
        case 'add':
            $result = $num1 + $num2;
            break;
        case 'subtract':
            $result = $num1 - $num2;
            break;
        case 'multiply':
            $result = $num1 * $num2;
            break;
        case 'divide':
            if ($num2 != 0) {
                $result = $num1 / $num2;
            } else {
                $result = "Error: Division by zero!";
            }
            break;
        default:
            $result = "Invalid operation!";
    }

    echo "<h3>Result: $result</h3>";
}
?>
</body>
</html>

File: Dart/calculator.dart
import 'dart:io';

void main() {
  print(
    "Choose your operation:\n1. Addition\n2. Subtraction\n3. Multiplication\n4. Division",
  );

  int? operation = int.tryParse(stdin.readLineSync() ?? '');
  if (operation == null || operation < 1 || operation > 4) {
    print("Invalid choice. Please enter a number between 1 and 4.");
    return;
  }

  print("First Number: ");
  num? a = num.tryParse(stdin.readLineSync() ?? '');
  if (a == null) {
    print("Invalid input. Please enter a valid number.");
    return;
  }

  print("Second Number: ");
  num? b = num.tryParse(stdin.readLineSync() ?? '');
  if (b == null) {
    print("Invalid input. Please enter a valid number.");
    return;
  }

  num result;
  switch (operation) {
    case 1:
      result = add(a, b);
      break;
    case 2:
      result = subtract(a, b);
      break;
    case 3:
      result = multiply(a, b);
      break;
    case 4:
      if (b == 0) {
        print("Error: Division by zero is not allowed.");
        return;
      }
      result = divide(a, b);
      break;
    default:
      print("Invalid operation.");
      return;
  }

  print("Result: $result");
}

num add(num a, num b) => a + b;
num subtract(num a, num b) => a - b;
num multiply(num a, num b) => a * b;
num divide(num a, num b) => a / b;
